## Visualize the computation using Tensorboard, simple example

In [1]:
import tensorflow as tf

In [3]:
# Define Placeholders in Tensorflow
X = tf.placeholder(tf.float32, name='X')
Y = tf.placeholder(tf.float32, name = 'Y')

In [5]:
Multiply = tf.multiply(X,Y)

    Now we are creating the session under which we have to run Multiply defined above
    Note: We have defined X and Y as place holders unlike constant, so we have provide or assign some value to do the math, :) Let's go

In [8]:
with tf.Session() as sess:
    result_mul = sess.run(Multiply, feed_dict={X:[10,21,4,2], Y:[12,4,22,4]})
    # Here it gets interesting, we have to add a new line to create a file which document the whole event.
    #I have chosen this directory, you can choose any
    event = tf.summary.FileWriter("logs/multiply-num", sess.graph)
    print(result_mul)
    

[ 120.   84.   88.    8.]


## Checkout the file at the destination with Tensorboard command, you can run terminal directly from Jupyter Notebook.

In [ ]:
!Tensorboard --logdir=logs

TensorBoard 0.1.5 at http://Tanays-MacBook-Air.local:6006 (Press CTRL+C to quit) 

## The above is the command to run this file, just go to the url, you will see the tensorboard with your computational graph. In case you are wondering, from where did local:6006 came from, as you might know, tensorflow is from Google, so it's basically indicates the ticker of Google, that is: GOOG---> 6006